applications/nrf_desktop/doc/dvfs.rst

```diff
@@ -0,0 +1,90 @@
+   The default value is 1 ms.
+#. :option:`CONFIG_DESKTOP_DVFS_RETRY_INIT_TIMEOUT_MS` - Timeout in milliseconds specifying time after which DVFS module will retry DVFS frequency change.
+   This timeout is applied in case DVFS is not yet initialized which causes the current request to fail.
+   The default value is 500 ms.
+#. :option:`CONFIG_DESKTOP_DVFS_RETRY_COUNT` - Number of retries of DVFS frequency change after which DVFS module will report MODULE_STATE_ERROR.
+   The default value is 5.
+
+Each DVFS state can be configured using the following Kconfig options:
+
+1. `CONFIG_DESKTOP_DVFS_STATE_<STATE>_ENABLE` - Enable DVFS state <STATE>.
+#. `CONFIG_DESKTOP_DVFS_STATE_<STATE>_TIMEOUT_MS` - Timeout in milliseconds specifying time after which the DVFS module returns to low frequency after entering <STATE>.
+#. `CONFIG_DESKTOP_DVFS_STATE_<STATE>_ACTIVE_FREQ` - Default active frequency of <STATE>.
+
+Currently supported DVFS states are:
+
+1. DVFS_STATE_INITIALIZING - State indicating that core initialization is in process.
+#. DVFS_STATE_LLPM_CONNECTED - State indicating that Low Latency Packet Mode is active.
+#. DVFS_STATE_USB_CONNECTED - State indicating that USB is connected.
+#. DVFS_STATE_CONFIG_CHANNEL - State indicating that Config Channel is active.
+#. DVFS_STATE_SMP_TRANSFER - State indicating that SMP transfer is active.
+
+Implementation details
+**********************
+
+The Dynamic Voltage and Frequency Scaling (DVFS) has two state tracing entities.
+
+First `module_state` is used to track the state of the DVFS module.
+It states if the DVFS module is ready or did it encounters an error.
+
+Second `dfvs_requests_state_bitmask` is used to track separately the state of each DVFS state.
+Each DVFS state is represented by a bit in the bitmask.
+If the bit is set, the state is active.
+DVFS module listens for events that are sent and applies state activity according to the event.
+DVFS module then checks which states are active and sets the highest frequency of all active states.
+If no state is active, the module sets the lowest frequency: DVFS_FREQ_LOW.
+
+Each change request is retried if it fails.
+The module will retry the request up to the number of times specified by :option:`CONFIG_DESKTOP_DVFS_RETRY_COUNT`.
+Failing frequency change request can be caused by another frequency change request in progress or by the DVFS module being not yet initialized.
+If the frequency change request is in progress, the module will retry the request after a timeout specified by :option:`CONFIG_DESKTOP_DVFS_RETRY_BUSY_TIMEOUT_MS`.
+If the DVFS module is not yet initialized, the module will retry the request after a timeout specified by :option:`CONFIG_DESKTOP_DVFS_RETRY_INIT_TIMEOUT_MS`.
+If the DVFS frequency request returns an error different than `EBUSY` or `EAGAIN`, the module will report MODULE_STATE_ERROR and set `module_state` to `STATE_ERROR`.
```

Comment:
shouldn't it be
```
... ``-EBUSY`` or ``-EAGAIN`` ...
```
?

Also maybe link those error codes with the sentences that describes the behaviour of the change request function above?